azurelinuxagent/ga/monitor.py

```diff
@@ -58,25 +58,30 @@ def _operation_impl(self):
         #
         # Check the processes in the agent cgroup
         #
-        processes = CGroupConfigurator.get_instance().get_processes_in_agent_cgroup()
-
-        if processes is not None:
-            unexpected_processes = []
-
-            for (_, command_line) in processes:
-                if not CGroupConfigurator.is_agent_process(command_line):
-                    unexpected_processes.append(command_line)
-
-            # Report a small sample of unexpected processes
-            if len(unexpected_processes) > 0 and self._error_count < 5:
-                unexpected_processes.sort()
-                error = ustr(unexpected_processes)
-                if error != self._last_error:
-                    self._error_count += 1
-                    self._last_error = error
-                    message = "The agent's cgroup includes unexpected processes: {0}".format(error)
-                    logger.info(message)
-                    add_event(op=WALAEventOperation.CGroupsDebug, message=message)
+        processes_check_error = None
+        try:
+            processes = CGroupConfigurator.get_instance().get_processes_in_agent_cgroup()
+
+            if processes is not None:
+                unexpected_processes = []
+
+                for (_, command_line) in processes:
+                    if not CGroupConfigurator.is_agent_process(command_line):
+                        unexpected_processes.append(command_line)
+
+                if len(unexpected_processes) > 0:
+                    unexpected_processes.sort()
+                    processes_check_error = ustr(unexpected_processes)
+        except Exception as e:
+            processes_check_error = ustr(e)
+
+        # Report a small sample of errors
+        if processes_check_error != self._last_error and self._error_count < 5:
```

Comment:
I missed this in the previous PR, but I noticed we're not resetting the error count ever. I think we should reset it once  a day or something to also get newer errors that might occur

Reply:
it was intentional; there is no need for that, i just want a sample of possible errors